Set some proper paths to make cmake find our tools.
Original patch by Heiko Przybyl

The ebuild now adds an extra / at the end of $EPREFIX so that it is
never the empty string (so that CMAKE_SYSTEM_PREFIX_PATH remains
correct)

--- cmake-2.8.0/Modules/Platform/Darwin.cmake
+++ cmake-2.8.0/Modules/Platform/Darwin.cmake
@@ -213,19 +213,23 @@
 
 
  
-# default to searching for frameworks first
-SET(CMAKE_FIND_FRAMEWORK FIRST)
+# default to searching for frameworks last
+SET(CMAKE_FIND_FRAMEWORK LAST)
 # set up the default search directories for frameworks
 SET(CMAKE_SYSTEM_FRAMEWORK_PATH
+  @GENTOO_PORTAGE_EPREFIX@Frameworks
+  @GENTOO_PORTAGE_EPREFIX@usr/lib
   ~/Library/Frameworks
   /Library/Frameworks
   /Network/Library/Frameworks
   /System/Library/Frameworks)
 
-# default to searching for application bundles first
-SET(CMAKE_FIND_APPBUNDLE FIRST)
+# default to searching for application bundles last
+SET(CMAKE_FIND_APPBUNDLE LAST)
 # set up the default search directories for application bundles
 SET(CMAKE_SYSTEM_APPBUNDLE_PATH
+  @GENTOO_PORTAGE_EPREFIX@Applications
+  @GENTOO_PORTAGE_EPREFIX@usr/bin
   ~/Applications
   /Applications
   /Developer/Applications)
--- cmake-2.8.0/Modules/Platform/UnixPaths.cmake
+++ cmake-2.8.0/Modules/Platform/UnixPaths.cmake
@@ -33,6 +33,7 @@
 # search types.
 LIST(APPEND CMAKE_SYSTEM_PREFIX_PATH
   # Standard
+  @GENTOO_PORTAGE_EPREFIX@usr/local @GENTOO_PORTAGE_EPREFIX@usr @GENTOO_PORTAGE_EPREFIX@
   /usr/local /usr /
 
   # CMake install location
@@ -44,43 +45,39 @@
 
 # List common include file locations not under the common prefixes.
 LIST(APPEND CMAKE_SYSTEM_INCLUDE_PATH
-  # Windows API on Cygwin
-  /usr/include/w32api
-
-  # X11
-  /usr/X11R6/include /usr/include/X11
-
-  # Other
-  /opt/local/include /usr/pkg/include
-  /opt/csw/include /opt/include  
-  /usr/openwin/include
+  @GENTOO_PORTAGE_EPREFIX@usr/include
   )
 
 LIST(APPEND CMAKE_SYSTEM_LIBRARY_PATH
-  # Windows API on Cygwin
-  /usr/lib/w32api
-
-  # X11
-  /usr/X11R6/lib /usr/lib/X11
-
-  # Other
-  /opt/local/lib /usr/pkg/lib
-  /opt/csw/lib /opt/lib 
-  /usr/openwin/lib
+  @GENTOO_PORTAGE_GCCLIBDIR@/gcc
+  @GENTOO_PORTAGE_GCCLIBDIR@
+  @GENTOO_PORTAGE_EPREFIX@usr/lib64
+  @GENTOO_PORTAGE_EPREFIX@usr/lib32
+  @GENTOO_PORTAGE_EPREFIX@usr/lib
+  @GENTOO_PORTAGE_EPREFIX@lib
   )
 
 LIST(APPEND CMAKE_SYSTEM_PROGRAM_PATH
-  /usr/pkg/bin
+  @GENTOO_PORTAGE_EPREFIX@usr/bin
+  @GENTOO_PORTAGE_EPREFIX@bin
   )
 
 LIST(APPEND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES
+  @GENTOO_PORTAGE_GCCLIBDIR@/gcc
+  @GENTOO_PORTAGE_GCCLIBDIR@
+  @GENTOO_PORTAGE_EPREFIX@usr/lib64
+  @GENTOO_PORTAGE_EPREFIX@usr/lib32
+  @GENTOO_PORTAGE_EPREFIX@usr/lib
+  @GENTOO_PORTAGE_EPREFIX@lib
   /lib /usr/lib /usr/lib32 /usr/lib64
   )
 
 LIST(APPEND CMAKE_C_IMPLICIT_INCLUDE_DIRECTORIES
+  @GENTOO_PORTAGE_EPREFIX@usr/include
   /usr/include
   )
 LIST(APPEND CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES
+  @GENTOO_PORTAGE_EPREFIX@usr/include
   /usr/include
   )
 
